stock_picking_batch_creation/wizards/make_picking_batch.py

```diff
@@ -0,0 +1,404 @@
+        """Get the number of bins used by the picking.
+        If the pickings are grouped by partner, we compute the number of new bins
+        required to add the picking to the batch. To do so, each time we add a
+        picking for a partner, we sum the volume of the current picking to
+        the volume of the previous pickings for the same partner stored in
+        self._volume_by_partners. (the self._volume_by_partners is updated
+        with this new value). The number of bins required to store the new
+        picking is the difference between the number of bins required to store
+        the previous volume and the number of bins required to store the new
+        volume.
+
+        :param picking: picking to get the number of bins
+        """
+        nbr_bins = picking._get_nbr_bins_for_device(self._device)
+        if self.group_pickings_by_partner:
+            picking_volume = picking.volume
+            if not picking_volume:
+                # if a picking has no volume defined we use the volume per bin
+                # of the device by convention a picking without volume fill a complete
+                # bin
+                picking_volume = self._device.volume_per_bin
+            old_volume = self._volume_by_partners[picking.partner_id]
+            new_volume = picking_volume + old_volume
+            nbr_bins = math.ceil(new_volume / self._device.volume_per_bin) - math.ceil(
+                old_volume / self._device.volume_per_bin
+            )
+            self._volume_by_partners[picking.partner_id] = new_volume
+        return nbr_bins
+
+    def _add_picking(self, picking):
+        """Add a picking to the batch.
+        At the same time, we update the counters to decrement the remaining
+        resources available for the batch.
+
+        :param picking: picking to add to the batch
+        """
+        self._selected_picking_ids.append(picking.id)
+        self._remaining_weight -= picking.weight
+        self._remaining_nbr_picking_lines -= picking.nbr_picking_lines
+        nbr_bins = self._get_nbr_bins_for_picking(picking)
+        self._remaining_nbr_bins -= nbr_bins
+        self._previous_selected_picking = picking
+
+    def _apply_limits(self):
+        """
+        Once we get the first picking and the device to use, we look
+        for other pickings to add to the batch. We add pickings as long as the
+        conditions are respected.
+        """
+        picking = self._first_picking
+        while picking:
+            if self._remaining_nbr_bins == 0:
+                break
+            picking = self._get_additional_picking()
+            if picking:
+                self._add_picking(picking)
+
+    def _create_batch_values(self):
+        selected_pickings = self.env["stock.picking"].browse(self._selected_picking_ids)
+        return {
+            "picking_ids": [(6, 0, selected_pickings.ids)],
+            "user_id": self._first_picking.user_id.id,
```

Comment:
If we set a user_id in the creation wizard, the resulting batch should have that user
```suggestion
            "user_id": self.user_id.id,
```